Sources/BigInt/BigInt.swift

```diff
@@ -0,0 +1,816 @@
+//===--- BigInt.swift -----------------------------------------*- swift -*-===//
+//
+// This source file is part of the Swift Numerics open source project
+//
+// Copyright (c) 2019 - 2020 Apple Inc. and the Swift Numerics project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+public struct BigInt: SignedInteger {
+
+  public typealias Words = [UInt]
+
+  public private(set) var words: Words
+
+  @usableFromInline
+  internal init(words: Words) {
+    self.words = words
+  }
+
+  public init<T>(bitPattern source: T) where T: BinaryInteger {
+    words = Words(source.words)
```

Comment:
There is no requirement that `words` in a BinaryInteger must be minimal -- it may have unnecessary words at the end, especially for (large) FixedWidthIntegers. So the `words` array needs be normalized after initialization here, or things like Equatable/Hashable won't work correctly.